src/main/host/managed_thread.rs

```diff
@@ -687,6 +697,29 @@ impl ManagedThread {
             .map(|_| nix::unistd::Pid::from_raw(child_pid))
         };
 
+        // Write the serialized shmem descriptor to the stdin pipe. The pipe
+        // buffer should be large enough that we can write it all without having
+        // to wait for data to be read.
+        if child_pid_res.is_ok() {
+            // we avoid using the nix write wrapper here, since we can't guarantee
+            // that all bytes of the serialized shmem block are initd, and hence
+            // can't safely construct the &[u8] that the nix wrapper wants.
+            let serialized = shmem_block.serialize();
+            let serialized_bytes = shadow_pod::as_u8_slice(&serialized);
+            let written = nix::errno::Errno::result(unsafe {
+                libc::write(
+                    stdin_writer,
+                    serialized_bytes.as_ptr().cast(),
+                    serialized_bytes.len(),
+                )
+            })
+            .unwrap();
+            // TODO: loop if needed. Shouldn't be in practice, though.
+            assert_eq!(written, isize::try_from(serialized_bytes.len()).unwrap());
```

Comment:
If the pipe is blocking, Linux guarantees that the write will block until all `n` bytes have been written. So as long as the user hasn't set the max pipe size smaller than the struct size (in which case shadow would deadlock or maybe ENOSPC), this should be guaranteed (but no harm in checking the rv anyways).